cmd/csi-snapshotter/create_crd.go

```diff
@@ -65,6 +80,22 @@ func CreateCRD(clientset apiextensionsclient.Interface) error {
 			},
 		},
 	}
+
+	rep, err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Get(crd.Name, metav1.GetOptions{})
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			klog.Fatalf("failed to get VolumeSnapshotResource: %#v, err: %#v",
+				rep, err)
+		} else {
+			klog.Infof("can not find VolumeSnapshotResource: %s, err: %#v", crd.Name, err)
+		}
+	} else {
+		err = clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Delete(crd.Name, &metav1.DeleteOptions{})
```

Comment:
We must not use delete crd method, when we delete the crd, all the CR will also be deleted. we should compare the crd spec and update the crd if the spec changes.

Reply:
yes, if the spec not changes, no need to delete the crd

Reply:
Thanks @wackxu for your comments!
Hi @zhucan Can you try @wackxu's suggestion?  Check if the spec has changed first, and then use the Update method, not the Delete, to update the CRD.

Reply:
We should not install crd with the snapshotter and instead make it admin operation.

Reply:
@wackxu @xing-yang Thanks, it's a good suggestion

Reply:
@msau42 What you mean is to put 'install crd' step in admin operation, for example, create from yaml instead of install crd from snapshotter？